contrib/podex/podex.go

```diff
@@ -32,22 +32,29 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"log"
 	"net/http"
 	"os"
 	"strconv"
 	"strings"
 
-	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta1"
 	"github.com/ghodss/yaml"
+	goyaml "gopkg.in/v2/yaml"
 )
 
-const usage = "usage: podex [-json|-yaml] [-id PODNAME] IMAGES"
+const usage = "podex [-format=yaml|json] [-type=pod|container] [-id NAME] IMAGES..."
 
-var generateJSON = flag.Bool("json", false, "generate json manifest")
-var generateYAML = flag.Bool("yaml", false, "generate yaml manifest")
-var podName = flag.String("id", "", "set pod name")
+var manifestFormat = flag.String("format", "yaml", "manifest format to output, `yaml` or `json`")
+var manifestType = flag.String("type", "pod", "manifest type to output, `pod` or `container`")
+var manifestId = flag.String("id", "", "manifest id, default to image base name")
```

Comment:
I think "name" is a better name for this flag. We had some id/name conflation going on for a while, we're working on getting to pod name (user defined) and pod uid (system defined). "id" is no longer a thing.